introspection-engine/connectors/sql-introspection-connector/tests/unit_tests/postgres.rs

```diff
@@ -0,0 +1,525 @@
+        custom_assert(&result, dm);
+    });
+}
+
+#[test]
+fn introspecting_a_one_to_many_relation_should_work() {
+    test_each_backend_with_ignores(vec![SqlFamily::Sqlite, SqlFamily::Mysql], |test_setup, barrel| {
+        let _setup_schema = barrel.execute(|migration| {
+            migration.create_table("User", |t| {
+                t.add_column("id", types::primary());
+            });
+            migration.create_table("Post", |t| {
+                t.add_column("id", types::primary());
+                t.inject_custom("user_id INTEGER REFERENCES \"User\"(\"id\")");
+            });
+        });
+
+        let dm = r#"  
+            model Post {
+               id      Int @id(strategy: NONE) @sequence(name: "Post_id_seq", allocationSize: 1, initialValue: 1)
+               user_id User?
+            }
+            
+            model User {
+               id      Int @id(strategy: NONE) @sequence(name: "User_id_seq", allocationSize: 1, initialValue: 1)
+               posts Post[] 
+            }
+        "#;
+        let result = dbg!(introspect(test_setup));
+        custom_assert(&result, dm);
+    });
+}
+
+#[test]
+fn introspecting_a_one_req_to_many_relation_should_work() {
+    test_each_backend_with_ignores(vec![SqlFamily::Sqlite, SqlFamily::Mysql], |test_setup, barrel| {
+        let _setup_schema = barrel.execute(|migration| {
+            migration.create_table("User", |t| {
+                t.add_column("id", types::primary());
+            });
+            migration.create_table("Post", |t| {
+                t.add_column("id", types::primary());
+                t.inject_custom("user_id INTEGER NOT NULL REFERENCES \"User\"(\"id\")");
+            });
+        });
+
+        let dm = r#"
+            model Post {
+               id      Int @id(strategy: NONE) @sequence(name: "Post_id_seq", allocationSize: 1, initialValue: 1)
+               user_id User
+            }
+            
+            model User {
+               id      Int @id(strategy: NONE) @sequence(name: "User_id_seq", allocationSize: 1, initialValue: 1)
+               posts Post[] 
+            }
+       "#;
+        let result = dbg!(introspect(test_setup));
+        custom_assert(&result, dm);
+    });
+}
+
+#[test]
+fn introspecting_a_prisma_many_to_many_relation_should_work() {
+    test_each_backend_with_ignores(vec![SqlFamily::Sqlite, SqlFamily::Mysql], |test_setup, barrel| {
+        let _setup_schema = barrel.execute(|migration| {
+            migration.create_table("User", |t| {
+                t.add_column("id", types::primary());
+            });
+            migration.create_table("Post", |t| {
+                t.add_column("id", types::primary());
+            });
+            migration.create_table("_PostToUser", |t| {
+                t.inject_custom(
+                    "A INTEGER NOT NULL REFERENCES  \"User\"(\"id\"),
+                    B INTEGER NOT NULL REFERENCES  \"Post\"(\"id\")",
```

Comment:
We just talked about this: we are not sure whether the query engine will do the right thing if the two models are not lexicographically order (model A.name < model B.name). If it can't cope with it, we should generate a join table.

Reply:
Just tested it with Dom. It doesn't ;-) I'll adjust the introspection criteria to be more narrow.